app/jobs/regular/topic_action_converter.rb

```diff
@@ -0,0 +1,18 @@
+class Jobs::TopicActionConverter < Jobs::Base
+
+  # Re-creating all the user actions could be very slow, so let's do it in a job
+  # to avoid a N+1 query on a front facing operation.
+  def execute(args)
+    topic = Topic.find_by(id: args[:topic_id])
```

Comment:
I was wondering if an includes `:posts` will help us here. We only moved the N+1 issue to sidekiq

Reply:
By default in Rails that would not change anything, because it would select all the posts in the `.posts` method. The real N+1 concern here was that for every post we are creating a new user history object.